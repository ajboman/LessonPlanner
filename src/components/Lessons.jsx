import { Card, Button } from 'flowbite-react';
import { useEffect, useState } from 'react';
import LessonDelete from '../components/LessonDelete';

const Lessons = ({ allLessons, onDeleteLesson }) => {
  const [lessons, setLessons] = useState([]);
  const [showModal, setShowModal] = useState(false);
  const [lessonToDelete, setLessonToDelete] = useState(null);

  const handleDeleteLesson = (lessonId) => {
    onDeleteLesson(lessonId);
    setShowModal(false);  
  };

  const handleOpenModal = (lessonId) => {
    setLessonToDelete(lessonId);
    setShowModal(true);
  };

  useEffect(() => {
    setLessons(allLessons);
  }, [allLessons]);

  return (
    <div>
      <h1 className="mb-4 text-xl font-bold">Saved Lessons</h1>
      <div className="grid lg:grid-cols-4 md:grid-cols-3 sm:grid-cols-2 grid-cols-1 gap-4">
        {lessons.length === 0 ? (
          <p>No lessons saved yet.</p>
        ) : (
          lessons.map((lesson) => (
            <Card
              key={lesson.id}
              className="relative overflow-y-auto max-h-96"
              style={{ backgroundColor: 'transparent' }}
            >
              <Button
                onClick={() => handleOpenModal(lesson.id)}
                className="absolute top-2 right-2 bg-red-500 hover:bg-red-600 text-red-500 hover:text-white font-bold py-1 px-2 rounded"
                aria-label={`Delete lesson ${lesson.id}`}
                variant="danger"
                size="sm"
              >
                <span className="text-lg">âœ•</span>
              </Button>
              <div className="font-normal text-gray-900 dark:text-gray-400 pl-2 whitespace-pre-wrap">
                {lesson.lesson}
              </div>
            </Card>
          ))
        )}
      </div>
      <LessonDelete
        showModal={showModal}
        setShowModal={setShowModal}
        handleDeleteLesson={handleDeleteLesson}
        lessonToDelete={lessonToDelete}
      />
    </div>
  );
};

export default Lessons;
