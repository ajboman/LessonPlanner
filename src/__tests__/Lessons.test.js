import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Lessons from '../components/Lessons';

describe('Lessons component', () => {
  const mockLessons = [
    'Lesson 1',
    'Lesson 2',
    'Lesson 3',
  ];

  test('renders correctly', () => {
    const { container } = render(
      <Lessons allLessons={mockLessons} onDeleteLesson={() => {}} />
    );
    expect(container).toMatchSnapshot();
  });

  test('displays lessons correctly', () => {
    const { getByText } = render(
      <Lessons allLessons={mockLessons} onDeleteLesson={() => {}} />
    );

    mockLessons.forEach((lesson) => {
      const lessonElement = getByText(lesson);
      expect(lessonElement).toBeInTheDocument();
    });
  });

  test('calls onDeleteLesson when delete button is clicked', () => {
    const mockDeleteLesson = jest.fn();
    const { getByLabelText } = render(
      <Lessons allLessons={mockLessons} onDeleteLesson={mockDeleteLesson} />
    );

    mockLessons.forEach((lesson, index) => {
      const deleteButton = getByLabelText(`Delete lesson ${index}`);
      fireEvent.click(deleteButton);
      expect(mockDeleteLesson).toHaveBeenCalledWith(index);
    });
  });
});
