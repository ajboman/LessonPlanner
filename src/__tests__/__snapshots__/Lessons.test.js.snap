// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lessons component renders correctly 1`] = `
<div>
  <div>
    <h1
      class="mb-4 text-xl font-bold"
    >
      Saved Lessons
    </h1>
    <div
      class="grid lg:grid-cols-4 md:grid-cols-3 sm:grid-cols-2 grid-cols-1 gap-4"
    >
      <div
        class="flex rounded-lg border border-gray-200 bg-white shadow-md dark:border-gray-700 dark:bg-gray-800 flex-col relative overflow-y-auto max-h-96"
        data-testid="flowbite-card"
        style="background-color: transparent;"
      >
        <div
          class="flex h-full flex-col justify-center gap-4 p-6"
        >
          <button
            aria-label="Delete lesson lesson1"
            class="text-white bg-cyan-700 border border-transparent hover:bg-cyan-800 focus:ring-4 focus:ring-cyan-300 disabled:hover:bg-cyan-700 dark:bg-cyan-600 dark:hover:bg-cyan-700 dark:focus:ring-cyan-800 dark:disabled:hover:bg-cyan-600 focus:!ring-2 group flex h-min items-center justify-center p-0.5 text-center font-medium focus:z-10 rounded-lg absolute top-2 right-2 bg-red-500 hover:bg-red-600 text-red-500 hover:text-white font-bold py-1 px-2 rounded"
            type="button"
            variant="danger"
          >
            <span
              class="flex items-center rounded-md text-sm px-3 py-1.5"
            >
              <span
                class="text-lg"
              >
                ✕
              </span>
            </span>
          </button>
          <div
            class="font-normal text-gray-900 dark:text-gray-400 pl-2 whitespace-pre-wrap"
          />
        </div>
      </div>
      <div
        class="flex rounded-lg border border-gray-200 bg-white shadow-md dark:border-gray-700 dark:bg-gray-800 flex-col relative overflow-y-auto max-h-96"
        data-testid="flowbite-card"
        style="background-color: transparent;"
      >
        <div
          class="flex h-full flex-col justify-center gap-4 p-6"
        >
          <button
            aria-label="Delete lesson lesson2"
            class="text-white bg-cyan-700 border border-transparent hover:bg-cyan-800 focus:ring-4 focus:ring-cyan-300 disabled:hover:bg-cyan-700 dark:bg-cyan-600 dark:hover:bg-cyan-700 dark:focus:ring-cyan-800 dark:disabled:hover:bg-cyan-600 focus:!ring-2 group flex h-min items-center justify-center p-0.5 text-center font-medium focus:z-10 rounded-lg absolute top-2 right-2 bg-red-500 hover:bg-red-600 text-red-500 hover:text-white font-bold py-1 px-2 rounded"
            type="button"
            variant="danger"
          >
            <span
              class="flex items-center rounded-md text-sm px-3 py-1.5"
            >
              <span
                class="text-lg"
              >
                ✕
              </span>
            </span>
          </button>
          <div
            class="font-normal text-gray-900 dark:text-gray-400 pl-2 whitespace-pre-wrap"
          />
        </div>
      </div>
      <div
        class="flex rounded-lg border border-gray-200 bg-white shadow-md dark:border-gray-700 dark:bg-gray-800 flex-col relative overflow-y-auto max-h-96"
        data-testid="flowbite-card"
        style="background-color: transparent;"
      >
        <div
          class="flex h-full flex-col justify-center gap-4 p-6"
        >
          <button
            aria-label="Delete lesson lesson3"
            class="text-white bg-cyan-700 border border-transparent hover:bg-cyan-800 focus:ring-4 focus:ring-cyan-300 disabled:hover:bg-cyan-700 dark:bg-cyan-600 dark:hover:bg-cyan-700 dark:focus:ring-cyan-800 dark:disabled:hover:bg-cyan-600 focus:!ring-2 group flex h-min items-center justify-center p-0.5 text-center font-medium focus:z-10 rounded-lg absolute top-2 right-2 bg-red-500 hover:bg-red-600 text-red-500 hover:text-white font-bold py-1 px-2 rounded"
            type="button"
            variant="danger"
          >
            <span
              class="flex items-center rounded-md text-sm px-3 py-1.5"
            >
              <span
                class="text-lg"
              >
                ✕
              </span>
            </span>
          </button>
          <div
            class="font-normal text-gray-900 dark:text-gray-400 pl-2 whitespace-pre-wrap"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
